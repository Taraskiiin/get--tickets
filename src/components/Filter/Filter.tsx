import React from "react";
import { ButtonGroup, Button } from "react-bootstrap";
import styled from "styled-components";

const Filter = () => {
  return (
    <Nav className="me-2">
      <StyledButton href="#">Самый дешевый</StyledButton>
      <StyledButton href="#">Самый быстрый</StyledButton>
    </Nav>
  );
};

const StyledButton = styled(Button)`
  text-decoration: none;
  text-align: center;
  padding: 14px 0;
  font-size: 14px;
  color: #4a4a4a;
  background-color: #fff;
  border: 1px solid #333333;

  &:hover {
    color: #fff;
    background-color: #2196f3;
  }

  &:active {
    color: #fff;
    background-color: #2196f3;
  }
`;

const Nav = styled(ButtonGroup)`
  width: 100%;
  height: 50px;
  display: flex;
  justify-content: space-between;
  text-transform: uppercase;
  letter-spacing: 0.5px;
  border-radius: 10px;
  background-color: #fff;
  margin-bottom: 20px;
`;

export default Filter;
