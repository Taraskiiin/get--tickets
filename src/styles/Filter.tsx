import styled from 'styled-components';
import { ButtonGroup, Button } from 'react-bootstrap';

export const Nav = styled(ButtonGroup)`
  width: 100%;
  height: max-content;
  display: flex;
  justify-content: space-between;
  text-transform: uppercase;
  letter-spacing: 0.5px;
  border-radius: 10px;
  background-color: #fff;
  margin-bottom: 20px;
`;

export const StyledButton = styled(Button)`
  text-decoration: none;
  text-align: center;
  padding: 14px 0;
  font-size: 14px;
  color: #4a4a4a;
  background-color: #fff;
  border: 1px solid #333333;

  &:hover {
    color: #fff;
    background-color: #2196f3;
  }

  &:active {
    color: #fff;
    background-color: #2196f3;
  }
`;
