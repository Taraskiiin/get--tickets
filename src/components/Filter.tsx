import React from 'react';

import { useQueryParams } from '../hooks/useQueryParams';
import { Nav, StyledButton, Link } from '../styles/Filter';

const Filter: React.FC = () => {
  const queryParams = useQueryParams();

  const queryParamsState = {
    querySortBy: queryParams.get('sortBy'),
    queryPage: Number(queryParams.get('page')),
    queryStops: queryParams.get('stops'),
  };

  return (
    <Nav className="me-2">
      <Link
        to={`?sortBy=${'price'}&page=1&stops=${queryParamsState.queryStops}`}
      >
        <StyledButton active={queryParamsState.querySortBy === 'price'}>
          Самый дешевый
        </StyledButton>
      </Link>
      <Link
        to={`?sortBy=${'time'}&page=1&stops=${queryParamsState.queryStops}`}
      >
        <StyledButton active={queryParamsState.querySortBy === 'time'}>
          Самый быстрый
        </StyledButton>
      </Link>
    </Nav>
  );
};

export default Filter;
