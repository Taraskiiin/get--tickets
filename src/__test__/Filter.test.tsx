import React from 'react';
import Filter from '../components/Filter';
import renderWith from './RenderWith';

describe('<Filter /> testing', () => {
  it('Should render with Provider and ThemeProvider', () => {
    expect(renderWith.withRouterAndThemeProvider(<Filter />)).toMatchSnapshot();
  });
});
