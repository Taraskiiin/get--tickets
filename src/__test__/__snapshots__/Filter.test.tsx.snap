// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> testing should match a snapshot 1`] = `
<div
  className="sc-bdvvtL lmfyqF me-2 btn-group"
  role="group"
>
  <a
    aria-current="page"
    className="sc-gsDKAQ fNHwLb active"
    href="/?sortBy=price&page=1&stops=null"
    onClick={[Function]}
  >
    <button
      className="sc-dkPtRN gdWSvs btn btn-primary"
      disabled={false}
      type="button"
    >
      Самый дешевый
    </button>
  </a>
  <a
    aria-current="page"
    className="sc-gsDKAQ fNHwLb active"
    href="/?sortBy=time&page=1&stops=null"
    onClick={[Function]}
  >
    <button
      className="sc-dkPtRN gdWSvs btn btn-primary"
      disabled={false}
      type="button"
    >
      Самый быстрый
    </button>
  </a>
</div>
`;
